api/protogen/protogen_test.go

```diff
@@ -115,6 +115,98 @@ func (suite *GenerateTestSuite) TestTerraformAttributeTypeToProtoType() {
 				},
 			},
 		},
+		"object": {
+			tfType: types.ObjectType{
+				AttrTypes: map[string]attr.Type{
+					"nested": types.StringType,
+				},
+			},
+			expectedType: "TheField",
+			expectedMessage: &message{
+				Name: "TheField",
+				Fields: []field{
+					{
+						Type: "string",
+						Name: "nested",
+						Tag:  1,
+					},
+				},
+			},
+		},
+		"object-with-nested-object": {
+			tfType: types.ObjectType{
+				AttrTypes: map[string]attr.Type{
+					"name": types.StringType,
+					"phone_numbers": types.ObjectType{
+						AttrTypes: map[string]attr.Type{
+							"office": types.StringType,
+							"mobile": types.StringType,
+						},
+					},
+				},
+			},
+			expectedType: "TheField",
+			expectedMessage: &message{
+				Name: "TheField",
+				Messages: []message{
+					{
+						Name: "PhoneNumbers",
+						Fields: []field{
+							{
+								Type: "string",
+								Name: "mobile",
+								Tag:  1,
+							},
+							{
+								Type: "string",
+								Name: "office",
+								Tag:  2,
+							},
+						},
+					},
+				},
+				Fields: []field{
+					{
+						Type: "string",
+						Name: "name",
+						Tag:  1,
+					},
+					{
+						Type: "PhoneNumbers",
+						Name: "phone_numbers",
+						Tag:  2,
+					},
+				},
+			},
+		},
+		"list-nested-objects": {
+			tfType: types.ListType{
+				ElemType: types.ObjectType{
+					AttrTypes: map[string]attr.Type{
+						"email": types.StringType,
+						"name":  types.StringType,
+					},
+				},
+			},
+			expectedRepeated: true,
+			expectedType:     "TheField",
+			expectedMessage: &message{
+				Name:     "TheField",
+				Messages: nil,
+				Fields: []field{
+					{
+						Type: "string",
+						Name: "email",
+						Tag:  1,
+					},
+					{
+						Type: "string",
+						Name: "name",
+						Tag:  2,
+					},
+				},
+			},
+		},
 	}
 
```

Comment:
Can we add a test for List of objects where the objects are defined separately. 
Something like
```
var ExampleType = types.ObjectType{
	AttrTypes: map[string]attr.Type{
		"field1":   types.StringType,
		"field2": types.StringType,
	},
}

"example_list": resource_schema.ListAttribute{
					Optional:    true,
					Description: "list of Examples",
					ElementType: ExampleType,
				},
```

Reply:
tested